torchvision/prototype/transforms/functional/_meta_conversion.py

```diff
@@ -1,50 +1,91 @@
-        if old_format is None:
-            raise TypeError("For vanilla tensors the `old_format` needs to be provided.")
-        return box_convert(input, in_fmt=kwargs["old_format"].name.lower(), out_fmt=format.name.lower())
-    elif isinstance(input, features.BoundingBox):
-        output = K.convert_bounding_box_format(input, old_format=input.format, new_format=kwargs["format"])
-        return features.BoundingBox.new_like(input, output, format=format)
-
-    raise RuntimeError
-
-
-@dispatch(
-    {
-        torch.Tensor: None,
-        PIL.Image.Image: None,
-        features.Image: None,
-    }
-)
-def convert_color_space(input: Any, *args: Any, **kwargs: Any) -> Any:
-    color_space = kwargs["color_space"]
-    if type(input) is torch.Tensor or isinstance(input, PIL.Image.Image):
-        if color_space != features.ColorSpace.GRAYSCALE:
-            raise ValueError("For vanilla tensors and PIL images only RGB to grayscale is supported")
-        return _F.rgb_to_grayscale(input)
-    elif isinstance(input, features.Image):
-        output = K.convert_color_space(input, old_color_space=input.color_space, new_color_space=color_space)
-        return features.Image.new_like(input, output, color_space=color_space)
-
-    raise RuntimeError
+from torchvision.prototype.features import BoundingBoxFormat, ColorSpace
+from torchvision.transforms import functional_tensor as _FT, functional_pil as _FP
+
+
+def _xywh_to_xyxy(xywh: torch.Tensor) -> torch.Tensor:
+    xyxy = xywh.clone()
+    xyxy[..., 2:] += xyxy[..., :2]
+    return xyxy
+
+
+def _xyxy_to_xywh(xyxy: torch.Tensor) -> torch.Tensor:
+    xywh = xyxy.clone()
+    xywh[..., 2:] -= xywh[..., :2]
+    return xywh
+
+
+def _cxcywh_to_xyxy(cxcywh: torch.Tensor) -> torch.Tensor:
+    cx, cy, w, h = torch.unbind(cxcywh, dim=-1)
+    x1 = cx - 0.5 * w
+    y1 = cy - 0.5 * h
+    x2 = cx + 0.5 * w
+    y2 = cy + 0.5 * h
+    return torch.stack((x1, y1, x2, y2), dim=-1)
+
+
+def _xyxy_to_cxcywh(xyxy: torch.Tensor) -> torch.Tensor:
+    x1, y1, x2, y2 = torch.unbind(xyxy, dim=-1)
+    cx = (x1 + x2) / 2
+    cy = (y1 + y2) / 2
+    w = x2 - x1
+    h = y2 - y1
+    return torch.stack((cx, cy, w, h), dim=-1)
+
+
+def convert_bounding_box_format(
+    bounding_box: torch.Tensor, *, old_format: BoundingBoxFormat, new_format: BoundingBoxFormat
+) -> torch.Tensor:
+    if new_format == old_format:
+        return bounding_box.clone()
+
+    if old_format == BoundingBoxFormat.XYWH:
+        bounding_box = _xywh_to_xyxy(bounding_box)
+    elif old_format == BoundingBoxFormat.CXCYWH:
+        bounding_box = _cxcywh_to_xyxy(bounding_box)
+
+    if new_format == BoundingBoxFormat.XYWH:
+        bounding_box = _xyxy_to_xywh(bounding_box)
+    elif new_format == BoundingBoxFormat.CXCYWH:
+        bounding_box = _xyxy_to_cxcywh(bounding_box)
+
+    return bounding_box
+
+
+def _grayscale_to_rgb_tensor(grayscale: torch.Tensor) -> torch.Tensor:
+    return grayscale.expand(3, 1, 1)
```

Comment:
Non-blocking TODO: This kernel assumes the tensor is a single image and not a batch. We shouldn't make this assumption. We should fix on a separate PR.